Sources/ZippyJSON/ZippyJSONDecoder.swift

```diff
@@ -37,19 +38,15 @@ func isOnSimulator() -> Bool {
   #endif
 }
 
-public final class ZippyJSONDecoder {
+public final class ZippyJSONDecoder: Sendable {
     @available(*, deprecated, message: "This flag is deprecated because full-precision parsing speed is now on par with imprecise, so it will just always use full-precision")
-    public var zjd_fullPrecisionFloatParsing = true
+    public let zjd_fullPrecisionFloatParsing = true
+
+    @Synchronized
     private static var _zjd_suppressWarnings: Bool = false
     public static var zjd_suppressWarnings: Bool {
-        get {
-            return _zjd_suppressWarnings
-        }
-        set {
-            objc_sync_enter(self)
-            defer { objc_sync_exit(self) }
```

Comment:
I don't think it's safe to use this API from Swift, especially when combined with other primitives which callers of this API may be using. Since I had to make other mutable properties thread-safe, I just used the same primitive for all of them

Reply:
interesting, i'd be curious to see any resources you have indicating it's unsafe, although using a lock does seem simpler